pkg/policy/api/rule_validation.go

```diff
@@ -359,17 +360,25 @@ func (pr *PortRule) sanitize(ingress bool) error {
 		if isZero {
 			haveZeroPort = true
 		}
-		hasDNSRules := pr.Rules != nil && len(pr.Rules.DNS) > 0
 		// DNS L7 rules can be TCP, UDP or ANY, all others are TCP only.
 		switch {
 		case pr.Rules.IsEmpty(), hasDNSRules:
 			// nothing to do if no rules OR they are DNS rules (note the comma above)
 		case pr.Ports[i].Protocol != ProtoTCP:
 			return fmt.Errorf("L7 rules can only apply to TCP (not %s) except for DNS rules", pr.Ports[i].Protocol)
 		}
+	}
+
+	if ingress && hasDNSRules {
```

Comment:
Couldn't this logic be moved above the for-loop?

Reply:
yes! Done.